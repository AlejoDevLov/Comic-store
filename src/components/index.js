

export * from './Header';
export * from './Shop';
export * from './MenuCarList';
export * from './Article';
export * from './FilterMenu';
export * from './SizeTShirt';
export * from './HelpComponent';
export * from './IconsNavbar';