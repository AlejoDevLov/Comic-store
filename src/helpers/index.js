

export * from './allAnimes';
export * from './getItemById';
export  * from './setIconHelp';
