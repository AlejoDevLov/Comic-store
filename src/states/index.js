

export * from './carState';
export * from './listCarState';
export * from './shopListState';