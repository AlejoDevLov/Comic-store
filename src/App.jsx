import { Navigate, Route, Routes } from 'react-router-dom';
import HomePage from './pages/HomePage';
import { ShopPage, ArticlePage  } from './pages';
import { CarProvider } from './states/carState';


export const App = () => {
  return (
    <CarProvider>
      <Routes>
        <Route path='/home' element={ <HomePage /> }/>      
        <Route path='/*' element={ <Navigate to="/home"/> }/>
        <Route path='/shop/:anime' element={ <ShopPage/> } />
        <Route path='/articlePage/:itemName' element={ <ArticlePage /> }/>
      </Routes>
    </CarProvider>
  )
}
