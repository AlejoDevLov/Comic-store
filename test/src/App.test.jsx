import { render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { App } from "../../src/App"

describe( 'Pruebas en el componente App y sus rutas', () => {

    test('Debe renderizar la HomePage', () => {
        render(
            <MemoryRouter initialEntries={['/home']}>
                <App/>
            </MemoryRouter>
        )
        // screen.debug();
        expect(screen.getByText('ANIMECO SHOP')).toBeTruthy();
    });

    test( 'Debe hacer match con el snapshot en HomePage', () => {
        const { container } = render(
            <MemoryRouter initialEntries={['/home']}>
                <App/>
            </MemoryRouter>
        )
        // screen.debug();
        expect(container).toMatchSnapshot();
    });

    test('Debe renderizar el componente ShopPage', () => {
        const {container} = render(
                <MemoryRouter initialEntries={['/shop/dragon-ball']}>
                    <App/>
                </MemoryRouter>
                )
        // screen.debug();
        const nav = container.querySelector('nav');
        expect(nav).toBeTruthy();
    })
})