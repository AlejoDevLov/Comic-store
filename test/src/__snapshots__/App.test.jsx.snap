// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente App y sus rutas Debe hacer match con el snapshot en HomePage 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title-div"
    >
      <h1
        class="store-name"
      >
        ANIMECO SHOP
      </h1>
    </div>
    <div
      class="swiper swiper-coverflow swiper-3d swiper-initialized swiper-horizontal swiper-watch-progress mySwiper"
    >
      <div
        class="swiper-wrapper"
        style="cursor: grab; transition-duration: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          <a
            href="/shop/los-simpsons"
          >
            <img
              alt="los simpsons"
              src="/images/los-simpsom/1ca1e142f76931e4350d8a7c0501ec24.jpg"
            />
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            data-testid="demon"
            href="/shop/demon-slayer"
          >
            <img
              alt="demon-slayer"
              src="/images/demon/a1fe94673eb48244d0648c63fcaef0d4.jpg"
            />
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            href="/shop/dragon-ball"
          >
            <img
              alt="dragon ball"
              src="/images/dragon-ball/1a2bcec7310c0abc847cec52b5b65522.jpg"
            />
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            href="/shop/mario"
          >
            <img
              alt="mario"
              src="/images/mario/faec054c4f79e5d0d4e6b64d17dca593.jpg"
            />
          </a>
        </div>
        <div
          class="swiper-slide"
        >
          <a
            href="/shop/sevenDeadly"
          >
            <img
              alt="seven-deadly-sins"
              src="/images/the-seven-deadly/94d9cc928a2425acf71726c2fdc17d57.jpg"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
